library/core/src/intrinsics.rs

```diff
@@ -2686,11 +2700,11 @@ pub const unsafe fn const_deallocate(_ptr: *mut u8, _size: usize, _align: usize)
 // (`transmute` also falls into this category, but it cannot be wrapped due to the
 // check that `T` and `U` have the same size.)
 
-/// Check that the preconditions of an unsafe function are followed, if debug_assertions are on,
-/// and only at runtime.
+/// Check that the preconditions of an unsafe function are followed, at runtime if debug assertions
+/// are enabled, and in const-eval/Miri if the precondition is for library UB.
 ///
 /// This macro should be called as
-/// `assert_unsafe_precondition!((expr => name: Type, expr => name: Type) => Expression)`
+/// `assert_unsafe_precondition!(check_{library,lang}_ub, (expr => name: Type, expr => name: Type) => Expression)`
 /// where each `expr` will be evaluated and passed in as function argument `name: Type`. Then all
```

Comment:
That doesn't seem to match the actual syntax?

Reply:
Hah! Oli convinced me to update the syntax and I never kept the docs in sync.